src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -2011,112 +2011,135 @@ private long PurgeCounters(long upto, DocumentsOperationContext context, Collect
             if (tombstonesTable == null || tombstonesTable.NumberOfEntries == 0 || numberOfEntriesToDelete <= 0)
                 return 0;
 
-            var table = GetCountersTable(context.Transaction.InnerTransaction, collectionName);
-            if (table == null || table.NumberOfEntries == 0)
-                return 0;
-
+            Dictionary<string, HashSet<string>> countersToDelete = new Dictionary<string, HashSet<string>>(StringComparer.OrdinalIgnoreCase);
             var deleted = 0L;
+            var toDelete = 0L;
 
             foreach (var item in GetCounterTombstonesFrom(context, 0, upto))
```

Comment:
do we need to dispose this item? you moved to string in the dict, so I think we should dispose this?